src/bluesearch/entrypoint/database/run.py

```diff
@@ -0,0 +1,572 @@
+
+    def requires(self) -> luigi.Task:
+        """Define conditional dependencies."""
+        if self.source in {"pmc"}:
+            return self.clone(UnzipTask)
+        else:
+            return self.clone(DownloadTask)
+
+    def output(self) -> luigi.LocalTarget:
+        """Define output file path."""
+        input_dir = self.input()
+        output_file = Path(input_dir.path).parent / "topic_infos.jsonl"
+
+        return luigi.LocalTarget(str(output_file))
+
+    def program_args(self) -> list[str]:
+        """Define subprocess arguments."""
+        input_dir = self.input().path
+        output_dir = self.output().path
+
+        command = [
+            *BBS_BINARY,
+            "topic-extract",
+            *VERBOSITY,
+            self.source,
+            input_dir,
+            output_dir,
+        ]
+
+        if self.source in {"medrxiv", "biorxiv"}:
+            command.extend(
+                ["-R", "-m", r".*\.meca$"],
+            )
+
+        if self.source in {"pmc", "pubmed"}:
+            command.append(f"--mesh-topic-db={self.mesh_topic_db}")
+
+        return command
+
+
+@requires(TopicExtractTask)
+class TopicFilterTask(ExternalProgramTask):
+    """Run topic filtering entrypoint.
+
+    It inputs `topic_infos.jsonl` and `filter_config` (rules) and it
+    generates a file `filtering.csv`.
+    """
+
+    filter_config = luigi.Parameter()
+
+    def output(self) -> luigi.LocalTarget:
+        """Define output file."""
+        output_file = Path(self.input().path).parent / "filtering.csv"
+
+        return luigi.LocalTarget(str(output_file))
+
+    def program_args(self) -> list[str]:
+        """Define subprocess arguments."""
+        extracted_topics = self.input().path
+        output_file = self.output().path
+
+        command = [
+            *BBS_BINARY,
+            "topic-filter",
+            *VERBOSITY,
+            extracted_topics,
+            self.filter_config,
+            output_file,
+        ]
+
+        return command
+
+
+@requires(TopicFilterTask)
+class PerformFilteringTask(luigi.Task):
+    """Create folder that only contains relevant articles.
+
+    We only consider those articles that made it through the topic-filtering
+    stage. The only input is the `filtering.csv`.
+    """
+
+    def output(self) -> luigi.LocalTarget:
+        """Define output folder."""
+        output_dir = Path(self.input().path).parent / "filtered"
+
+        return luigi.LocalTarget(str(output_dir))
+
+    def run(self) -> None:
+        """Create symlinks."""
+        output_dir = Path(self.output().path)
+
+        filtering = pd.read_csv(self.input().path)
+
+        output_dir.mkdir(exist_ok=True)
+
+        if self.source == "pubmed":
+            # Find all input files (.xml.gz)
+            all_input_files = [Path(p) for p in filtering["path"].unique()]
+
+            # Iteratively Load each  of the files in memory
+            for input_file in all_input_files:
```

Comment:
IMO it should be fine since these paths are coming from the `filtering.csv` table. And it was generated by `topic-filter` so I would assume that are valid paths.

Reply:
True! :)